code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,58 @@
+/mob/living/human/proc/shadowling_hatch()
+	set category = "Shadowling"
+	set name = "Hatch"
+
+	usr.verbs -= /mob/living/human/proc/shadowling_hatch
+	switch(alert("Are you sure you want to hatch? You cannot undo this!",,"Yes","No"))
+		if("No")
+			usr << "<span class='warning'>You decide against hatching for now."
+			usr.verbs += /mob/living/human/proc/shadowling_hatch
+			return
+		if("Yes")
+			usr.Stun(1000) //nigga you aint movin
+			new /obj/structure/chrysalis(usr.loc)
+			usr.alpha = 0 //I am the invisible man..
+			usr.visible_message("<span class='warning'>[usr] suddenly bends over and vomits a purple substance, which rapidly forms into an egg-like structure around their body!</span>", \
+								"You begin creating a suitable home for you as you hatch.")
+
+			for(var/obj/item/I in usr) //drops all items
+				usr.unEquip(I)
+
+			sleep(50)
+			usr.visible_message("<span class='warning'><b>A chrysalis forms around [usr] and seals shut, closing them completely inside.</b></span>", \
+							"You have formed your chrysalis. In one full minute, your hatching will be complete.")
+.
+
+			sleep(180)
+			usr.visible_message("<span class='warning'>Something thumps against the side of the chrysalis.</span>", \
+							"<i>Your body jolts with a thump against the chrysalis' wall. Your heart begins to beat faster.</i>")
+
+			sleep(300)
+			usr.visible_message("<span class='warning'><b>Something inside of the chrysalis gurgles. Something begins tearing from within.</b></span>", \
+							"<i>Your false skin slips away and your true form begins its exit.</i>")
+
+			sleep(10)
+			usr << "<i><b>You rip and tear.</b></i>"
+
+			sleep(10)
+			usr << "<i><b>The chrysalis falls like water before you.</b></i>"
+
+			sleep(10)
+			usr << "<i><b>You are free!</b></i>"
+
+			sleep(90)
+			usr.alpha = 255
+			usr.SetStunned(0)
+			usr << "<i><b><font size=3>YOU LIVE!!!</i></b></font>"
+
+			usr.equip_to_slot_or_del(new /obj/item/clothing/under/shadowling(usr), slot_w_uniform)
+			usr.equip_to_slot_or_del(new /obj/item/clothing/shoes/shadowling(usr), slot_shoes)
+			usr.equip_to_slot_or_del(new /obj/item/clothing/suit/space/shadowling(usr), slot_wear_suit)
+			usr.equip_to_slot_or_del(new /obj/item/clothing/head/shadowling(usr), slot_head)
+			usr.equip_to_slot_or_del(new /obj/item/clothing/gloves/shadowling(usr), slot_gloves)
+			usr.equip_to_slot_or_del(new /obj/item/clothing/mask/gas/shadowling(usr), slot_wear_mask)
+			usr.equip_to_slot_or_del(new /obj/item/clothing/glasses/night/shadowling(usr), slot_glasses)
```

Comment:
I know this is basically the best way to do this, but it's also awful and I hate it.